GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/DehydrateTests.cs

```diff
@@ -157,12 +155,13 @@ public void DehydrateShouldFailOnWrongDiskLayoutVersion()
 
         private void DirectoryShouldContain(string directory, params string[] fileOrFolders)
         {
-            string[] folderItems = this.fileSystem.EnumerateDirectory(directory).Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries);
-            folderItems.Length.ShouldEqual(fileOrFolders.Length);
-            for (int i = 0; i < fileOrFolders.Length; i++)
-            {
-                Path.GetFileName(folderItems[i]).ShouldEqual(fileOrFolders[i]);
-            }
+            IEnumerable<string> onDiskItems =
+                this.fileSystem.EnumerateDirectory(directory)
+                .Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries)
+                .Select(path => Path.GetFileName(path))
+                .OrderByDescending(x => x);
+
+            onDiskItems.ShouldMatchInOrder(fileOrFolders.OrderByDescending(x => x));

```

Comment:
Are we trying to verify that folderItems & fileOrFolders are equivalent?
If so this works it's just a shame you have to do the sorting.

Would this be a reasonable alternative?
`Assert.IsTrue(new HashSet<string>(onDiskItems).SetEquals(fileOrFolders));`

Reply:
> Are we trying to verify that folderItems & fileOrFolders are equivalent?

Yep the issue here was that the order of the items in the folder is not the same on Windows and Mac.

> Would this be a reasonable alternative?

I had considered doing something similar to what you proposed, but what I liked about this option is that by using `ShouldMatchInOrder` we get nice output if the entries are not equal:

```
            if (groupList.Count != expectedValuesList.Count)
            {
                errorMessage.AppendLine(string.Format("{0} counts do not match. was: {1} expected: {2}", message, groupList.Count, expectedValuesList.Count));
            }

            foreach (T groupExtraItem in groupExtraItems)
            {
                errorMessage.AppendLine(string.Format("Extra: {0}", groupExtraItem));
            }

            foreach (T groupMissingItem in groupMissingItems)
            {
                errorMessage.AppendLine(string.Format("Missing: {0}", groupMissingItem));
            }

            if (errorMessage.Length > 0)
            {
                Assert.Fail("{0}\r\n{1}", message, errorMessage);
            }
```

However, just now I realized that `ShouldMatchInOrder` is not actually checking that they match in order so I'll push a commit to fix that.

Reply:
> However, just now I realized that ShouldMatchInOrder is not actually checking that they match in order so I'll push a commit to fix that.

This may be fun and find other bugs in the code :-)

Since it gives us the nice output I'm find with using it.  Alternatively if you wanted to go a bit more crazy refactoring
`        public static IEnumerable<T> ShouldMatch<T>(this IEnumerable<T> group, params Action<T>[] itemCheckers, bool inOrder = false)
`
may be another option